lib/basedriver/driver.js

```diff
@@ -241,11 +249,13 @@ class BaseDriver extends Protocol {
 
   /**
    * Test createSession inputs to see if this is a W3C Session or a MJSONWP Session
+   * TODO this exists here as an alias since the functionality was moved to
+   * lib/protocol, but external packages depend on its existence here. So those
+   * should be updated to rely on the export from lib/protocol before this is
+   * removed.
    */
-  static determineProtocol (desiredCapabilities, requiredCapabilities, capabilities) {
-    return _.isPlainObject(capabilities) ?
-      BaseDriver.DRIVER_PROTOCOL.W3C :
-      BaseDriver.DRIVER_PROTOCOL.MJSONWP;
+  static determineProtocol (...args) {
+    return determineProtocol (...args);
```

Comment:
isn't it a recursive call?

Reply:
it would be if it were `return this.determineProtocol(...args)` but it is calling the imported method, not itself. that's the point--i didn't want to make this a breaking change so i didn't want to remove the method from the BaseDriver class, but i needed the functionality itself to be defined elsewhere. so it is essentially an alias until we can move all the usages of it throughout the appium projects, then we can remove this method entirely.

Reply:
Why not make the breaking change and do the work to update the other packages?